integration/setup.go

```diff
@@ -0,0 +1,161 @@
+package integration
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/xichen2020/eventdb/services/eventdb/serve"
+	"github.com/xichen2020/eventdb/storage"
+
+	xconfig "github.com/m3db/m3x/config"
+	"github.com/m3db/m3x/instrument"
+	"github.com/m3db/m3x/log"
+	"github.com/uber-go/tally"
+)
+
+const (
+	gracefulShutdownTimeout = 15 * time.Second
+)
+
+var (
+	logger = log.NullLogger
+
+	// basePath contains the path to this directory.
+	// NB(wjang): When running the integration tests via `test-ci-integration`, the test binaries are
+	// prebuilt and run. The issue is that we rely on relative paths to the config and testdata files
+	// for our tests; these relative paths break when we run the binary tests. To get around this,
+	// at time of compilation, we save the path to the source code so that we can reference it
+	// when running the tests. This is hacky as fuck but works!
+	// Alternative was to hard code the configs in code but that's not possible as most of the configs
+	// are private.
+	basePath string
+)
+
+type testServerSetup struct {
+	db   storage.Database
+	opts instrument.Options
+	cfg  configuration
+
+	// Signals.
+	doneCh   chan struct{}
+	closedCh chan struct{}
+}
+
+func newTestServerSetup(t *testing.T, configFile string) *testServerSetup {
```

Comment:
You can pass in the test config string here instead of the file path. All the file loading logic does is just reading the file and do
```
unmarshal := yaml.UnmarshalStrict
		if opts.DisableUnmarshalStrict {
			unmarshal = yaml.Unmarshal
		}
		if err := unmarshal(data, config); err != nil {
			return err
		}
return validator.Validate(config)
```
which you can probably create a function `loadConfig(str string)` to do so.